commons/src/main/java/org/frankframework/util/StringResolver.java

```diff
@@ -170,8 +170,8 @@ private static String extractNextKey(String val, Map<?, ?> props1, Map<?, ?> pro
 	}
 
 	private static String extractNextExpression(String val, SubstitutionContext ctx) {
-		if (val.contains(VALUE_SEPARATOR)) {
-			ctx.tail = val.indexOf(VALUE_SEPARATOR);
+		if (val.substring(ctx.pointer).contains(VALUE_SEPARATOR)) {
+			ctx.tail = val.substring(ctx.pointer).indexOf(VALUE_SEPARATOR) + ctx.pointer;
```

Comment:
```suggestion
		int nextSeparatorPos = val.indexOf(VALUE_SEPARATOR, ctx.pointer);
		if (nextSeparatorPos >= 0) {
			ctx.tail = nextSeparatorPos;
```

Dit is wat efficienter omdat je niet 2x de substring vanaf positie berekent, en ook niet eerst de "contains" doet en daarna nog een keer opnieuw gaat zoeken.

Ik vind het daarmee ook duidelijker wat er gebeurt.

Reply:
Zeg dat wel